defrag/modules/search.py

```diff
@@ -0,0 +1,36 @@
+from importlib import import_module
+from defrag.modules import ALL_MODULES
+from defrag.modules.helpers import Query, QueryResponse
+from typing import List
+from pydantic import BaseModel
+from defrag import app
+from defrag.modules.helpers.services_manager import ServicesManager
+import asyncio
+
+__MOD_NAME__ = "global_search"
+
+search_map = {f: getattr(m, "search") for f, m in [(n, import_module(
+    f"defrag.modules.{n}")) for n in ALL_MODULES] if hasattr(m, "search")}
+
+
+class SearchQuery(BaseModel):
+    keywords: str
+    scope: List[str]
+
+
+@app.get(f"/{__MOD_NAME__}/")
+async def global_search(sq: SearchQuery) -> QueryResponse:
```

Comment:
We can not use a `SearchQuery` as a parameter here, it has to be a string with the keyword.

Reply:
> We can not use a `SearchQuery` as a parameter here, it has to be a string with the keyword.

Then I have not understood https://fastapi.tiangolo.com/tutorial/body/#declare-it-as-a-parameter corretly...

Reply:
lmao you are right, it felt so natural, and apparently there is a debate about this here: https://github.com/tiangolo/fastapi/issues/318. 

Fixed as of latest commit.